### DISCUSSION TITLE
How can I use nvim cmp instead omni completion in DAP repl?

### DISCUSSION THREAD

**Discussion by MironPascalCaseFan** [mfussenegger/nvim-dap#1397] [Q&A]
How can I use [nvim-cmp](https://github.com/hrsh7th/nvim-cmp) instead of omni-completion in the DAP REPL? Currently, I have to press Ctrl + X, Ctrl + O to open the built-in Vim completion popup, but I would prefer the popup to appear automatically after I type "." in the DAP REPL, allowing me to select the desired entry using my familiar nvim-cmp keymaps.

**Comment by Kailian-Jacy** [comment]
Seems like it's triggering `omnifunc` set by nvim-dap:

https://github.com/mfussenegger/nvim-dap/blob/881f7dc17c0b4204a1d844732f95539575350e5b/lua/dap/repl.lua#L554

Maybe you can customize a cmp source and set buf local omnifunc to trigger nvim-cmp. But personally I'll just turn off the auto completion.

**Reply by MironPascalCaseFan** [reply]
The main thing that tripped me up was nvim-cmp's default rule that disables autocompletion in a prompt buffer.
I modified this rule so it works when the prompt buffer is the dap repl and installed [this plugin](https://github.com/rcarriga/cmp-dap). After that, if the debug adapter has completion capability, then everything works.

I think it is possible to achieve the same with the omnifunc provided by nvim-dap (without cmp-dap) by adding an omnifunc source to nvim-cmp and modifying the rule I mentioned before, but I didn't test it.
